src/libcore/num/flt2dec/mod.rs

```diff
@@ -410,8 +410,8 @@ fn determine_sign(sign: Sign, decoded: &FullDecoded, negative: bool) -> &'static
 /// it will only print given digits and nothing else.
 ///
 /// The byte buffer should be at least `MAX_SIG_DIGITS` bytes long.
-/// There should be at least 5 parts available, due to the worst case like
-/// `[+][0.][0000][45][0000]` with `frac_digits = 10`.
+/// There should be at least 4 parts available, due to the worst case like
+/// `[+][0.][0000][2][0000]` with `frac_digits = 10`.
```

Comment:
My understanding of the use of the numbers here was that they were denoting the index of the associated `Part` in the `parts` array.  So the `45` shouldn't be understood as the number "45", but `parts[4]` and `parts[5]`.  (Which is not correct in the first place, I believe.)  You can see this notation on display a little more clearly in `to_shortest_exp_str`, where we have `[+][1][.][2345][e][-][6]` as a result of this patch, though note that 1-based indexing is being used here rather than 0-based...  I changed `45` to `2` following said understanding, as the non-zero numbers there would be `parts[2]`.

I agree that this is not the best notation for explaining things!  Perhaps a separate pull request would be appropriate to make the documentation a little clearer?